middle_end/flambda2/types/structures/code_age_relation.ml

```diff
@@ -79,16 +79,21 @@ let join ~target_t ~resolver t1 t2 id1 id2 : _ Or_unknown.t =
   else
     let id1_to_root = all_ids_up_to_root ~resolver t1 id1 in
     let id2_to_root = all_ids_up_to_root ~resolver t2 id2 in
-    let shared_ids =
-      Code_id.Set.inter
-        (Code_id.Map.keys target_t)
-        (Code_id.Set.inter id1_to_root id2_to_root)
+    let shared_ids = Code_id.Set.inter id1_to_root id2_to_root in
+    let shared_ids_in_scope =
+      Code_id.Set.filter
+        (fun id ->
+          let is_imported =
```

Comment:
Maybe we should add an `is_imported` function to `Code_id` and `Var_within_closure` ?